### PR TITLE
feat(widgets): add emptyMessage customization to List widget

### LINKED ISSUE #1510: [feature] Add "Empty State" Customization to Data Widgets
## What problem does this solve?

When providing an empty data array to widgets like Select, Combobox, or List, the UI renders an empty box. We should provide an easy way to display a "No data found" placeholder.

## Expected Behavior
Developers can pass emptyMessage: "No files found" in the widget options. If the data array is empty, this text is printed centered (and optionally dimmed) inside the widget bounds.

## Proposed Solution
- Add emptyMessage?: string to the configuration options of list-based widgets.
- In _renderSelf(), if data.length === 0 && this.emptyMessage, calculate the center of the _getContentRect() and render the string using screen.writeString().
- Add unit tests validating the empty string output.

## Which package would this belong in?

 @termuijs/widgets

## GSSoC contributor?

- [x] Yes. You contribute under GSSoC 2026 and want to work on this.
- [ ] No. You are filing a suggestion.

### PR DESCRIPTION
**Pull request by realtushartyagi**
## Description

This PR adds an "Empty State" feature to the `@termuijs/widgets` `List` component. Previously, rendering a `List` with an empty data array resulted in a completely blank box, leaving users wondering if the widget was broken or just empty.

Developers can now pass an optional `emptyMessage` string (e.g., `"No files found"`) in the `ListProps`. When the items array is empty, this message is automatically centered horizontally and vertically inside the widget's content rect and styled with the `dim` attribute.

## Related Issue

Closes #1510

## Which package(s)?

@termuijs/widgets

## Type of Change

- [ ] 🐛 Bug fix (`type:bug`)
- [x] ✨ Feature (`type:feature`)
- [ ] 📝 Docs (`type:docs`)
- [ ] 🧪 Tests (`type:testing`)
- [ ] ♻️ Refactor (`type:refactor`)
- [ ] 🎨 Design / UX (`type:design`)
- [ ] ♿ Accessibility (`type:accessibility`)
- [ ] ⚡ Performance (`type:performance`)
- [ ] 🔧 DevOps / CI (`type:devops`)
- [ ] 🔒 Security (`type:security`)

## Checklist

- [x] ⭐ You starred the repo. The `needs-star` check blocks your merge otherwise.
- [x] Tests pass locally: `bun vitest run`
- [x] Build passes: `bun run build`
- [x] Typecheck passes: `bun run typecheck`
- [x] You read [`CONTRIBUTING.md`](./CONTRIBUTING.md).
- [x] Your PR title follows `type: short description`.
- [x] Widget state mutators call `markDirty()` (if your change affects rendering).
- [x] No new `any` types without an inline comment explaining why.
- [x] No unrelated refactors bundled into this PR.

## GSSoC 2026 Participation

- [x] You are a GSSoC 2026 contributor.
- Your GSSoC profile: `https://gssoc.girlscript.org/profile/f49aa65e-aab2-4e9f-99d6-2174f339ecfe`

## Screenshots / Recordings (UI changes)

*N/A*

## Notes for the Reviewer

Added a strict unit test suite (`renders emptyMessage when the list is empty`) in `List.test.ts` to ensure the correct text is written to the screen buffer when `items` is empty and `emptyMessage` is provided.

### CODE REVIEW

File: packages/widgets/src/input/List.ts [hunk 1]
```diff
@@ -129,12 +135,22 @@ export class List extends Widget {
...
-    protected _renderSelf(screen: Screen): void {
+       protected _renderSelf(screen: Screen): void {
         const rect = this._getContentRect();
         const { x, y, width, height } = rect;
         if (width <= 0 || height <= 0) return;
 
         const attrs = styleToCellAttrs(this._style);
+
+        // Render Empty State if no items exist
+        if (this._items.length === 0 && this._emptyMessage) {
```
Comment (realtushartyagi) [inline packages/widgets/src/input/List.ts:146]: I added this early-return block to handle the empty state rendering. It calculates the vertical and horizontal center of _getContentRect() and renders the truncated emptyMessage with dim: true. Returning early avoids running the rest of the list rendering loop (though it wouldn't crash, it saves unnecessary calculations).
